app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt

```diff
@@ -115,330 +129,177 @@ public class ConfirmedLessonTest {
     composeTestRule.onNodeWithTag("cancelButton").assertIsDisplayed().performClick()
-    Thread.sleep(300) // Wait for the dialog to show up
+
+    composeTestRule.waitUntil(20000) { composeTestRule.onNodeWithTag("cancelDialog").isDisplayed() }
 
     // Confirm the dialog
     composeTestRule.onNodeWithTag("cancelDialogConfirmButton").assertIsDisplayed().performClick()
 
+    composeTestRule.waitUntil(20000) {
+      composeTestRule.onNodeWithTag("cancelDialog").isNotDisplayed()
+    }
+
     // Check that the cancellation has been confirmed
     composeTestRule.onNodeWithTag("cancelDialog").assertIsNotDisplayed()
     verify(mockLessonRepository).updateLesson(any(), any(), any())
     verify(mockNavigationActions).navigateTo(Screen.HOME)
-  }*/
+  }
 
-  /*@Test
+  @Test
   fun confirmedLessonScreenLessonCancellation_StudentRequestedLesson() {
     lessonViewModel.selectLesson(studentRequestedLesson)
 
-    var isLocationChecked = false
-
-    composeTestRule.setContent {
-      ConfirmedLessonScreen(
-          listProfilesViewModel = listProfilesViewModel,
-          lessonViewModel = lessonViewModel,
-          navigationActions = mockNavigationActions,
-          onLocationChecked = { isLocationChecked = true })
-    }
     composeTestRule.waitForIdle()
 
-    composeTestRule.waitUntil(15000) {
-      // wait max 15 seconds for the map to load,
-      // as soon as the map is ready, the next line will be executed
-      isLocationChecked
-    }
-    isLocationChecked = false
-
     // Click on the "Cancel Lesson" button
     composeTestRule.onNodeWithTag("cancelRequestButton").assertIsDisplayed().performClick()
 
     // Check that the cancellation has been confirmed
     verify(mockLessonRepository).updateLesson(any(), any(), any())
     verify(mockNavigationActions).navigateTo(Screen.HOME)
-  }*/
+  }
 
-  /*@Test
+  @Test
   fun confirmedLessonScreenLessonCancellation_PendingLesson() {
     lessonViewModel.selectLesson(pendingTutorConfirmationLesson)
     listProfilesViewModel.setCurrentProfile(studentProfile)
 
-    var isLocationChecked = false
-
-    composeTestRule.setContent {
-      ConfirmedLessonScreen(
-          listProfilesViewModel = listProfilesViewModel,
-          lessonViewModel = lessonViewModel,
-          navigationActions = mockNavigationActions,
-          onLocationChecked = { isLocationChecked = true })
-    }
     composeTestRule.waitForIdle()
 
-    composeTestRule.waitUntil(15000) {
-      // wait max 15 seconds for the map to load,
-      // as soon as the map is ready, the next line will be executed
-      isLocationChecked
-    }
-    isLocationChecked = false
-
     // Click on the "Cancel Lesson" button
     composeTestRule.onNodeWithTag("deleteButton").assertIsDisplayed().performClick()
 
     // Check that the cancellation has been confirmed
     verify(mockLessonRepository).deleteLesson(any(), any(), any())
     verify(mockNavigationActions).navigateTo(Screen.HOME)
-  }*/
+  }
```

Comment:
As said by the coaches, it is not a good practice to add some commented code in the main branch, I understood that you had some merging issue with Philipp but maybe you could have store the test in the wiki instead of commented it.